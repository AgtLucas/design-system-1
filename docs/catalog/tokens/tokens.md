## Typography 

```table
span: 4
rows:
  - Token: font.size.smaller
    CSS: .font-size--smaller
    px: 12
    rem: 0.75
  - Token: font.size.small
    CSS: .font-size--small
    px: 14
    rem: 0.875
  - Token: font.size.medium
    CSS: .font-size--medium
    px: 16
    rem: 1
  - Token: font.size.large
    CSS: .font-size--large
    px: 20
    rem: 1.25
  - Token: font.size.larger
    CSS: .font-size--larger
    px: 24
    rem: 1.5
  - Token: font.size.largest
    CSS: .font-size--largest
    px: 28
    rem: 1.75
  - Token: font.size.pageTitle
    CSS: .font-size--page-title
    px: 28
    rem: 1.75
  - Token: font.size.sectionTitle
    CSS: .font-size--section-title
    px: 24
    rem: 1.5
  - Token: font.size.subsectionTitle
    CSS: .font-size--subsection-title
    px: 20
    rem: 1.25
```
