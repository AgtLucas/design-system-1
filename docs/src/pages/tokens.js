import React from "react";
import { Helmet } from "react-helmet";
import { Box, Title, Link } from "@nulogy/components";
import { Layout, Intro, DocSection } from "../components";

export default () => (
  <Layout>
    <Helmet>
      <title>Tokens</title>
    </Helmet>
    <Box
      bg="whiteGrey" p="x6" borderRadius={ 1 }
      mb="x6"
    >
      <Title mb="none">Tokens</Title>
      <Intro>Here you'll find all of the design options for creating interfaces in Nulogy's style.</Intro>
    </Box>

    <DocSection>
            Tokens can currently be found in the `/src/` folder of the <Link href="https://github.com/nulogy/design-system/tree/master/tokens/">@nulogy/tokens</Link> package.
    </DocSection>
  </Layout>
);
