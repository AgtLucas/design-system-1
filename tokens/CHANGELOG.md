# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.21.0](https://github.com/nulogy/design-system/compare/v0.20.0...v0.21.0) (2020-01-29)

**Note:** Version bump only for package @nulogy/tokens

# [0.20.0](https://github.com/nulogy/design-system/compare/v0.19.0...v0.20.0) (2020-01-20)

**Note:** Version bump only for package @nulogy/tokens

# [0.19.0](https://github.com/nulogy/design-system/compare/v0.18.2...v0.19.0) (2020-01-10)

**Note:** Version bump only for package @nulogy/tokens

# [0.18.0](https://github.com/nulogy/design-system/compare/v0.16.10...v0.18.0) (2020-01-06)

**Note:** Version bump only for package @nulogy/tokens

# [0.17.0](https://github.com/nulogy/design-system/compare/v0.16.10...v0.17.0) (2019-12-06)

**Note:** Version bump only for package @nulogy/tokens
