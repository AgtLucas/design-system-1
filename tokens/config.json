{
    "source": ["src/**/*.json"],
    "platforms": {
      "css": {   
        "transforms": ["attribute/cti", "name/cti/kebab", "size/px", "color/css"],
        "buildPath": "build/",
        "files": [{
          "destination": "variables.css",
          "format": "css/variables"
        }]
      },      
      "scss": {   
        "transforms": ["attribute/cti", "name/cti/kebab", "size/px", "color/css"],
        "buildPath": "build/",
        "files": [{
          "destination": "_variables.scss",
          "format": "scss/variables"
        }]
      },
      "react": {
        "transforms": ["attribute/cti", "name/cti/snake", "size/px", "color/hex"],
        "buildPath": "build/",
        "files": [{
          "destination": "exports.js",
          "format": "javascript/es6"
        }]
      }
    }
  }


