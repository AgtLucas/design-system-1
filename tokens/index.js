export const colours = {
  primary: 'coral',
  secondary: 'white'
};

export default {
  colours,
  borderRadius: '0.4rem'
};
