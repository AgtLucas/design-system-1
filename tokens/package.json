{
  "name": "@nulogy/tokens",
  "version": "0.1.1",
  "description": "Design tokens for the Nulogy Design System - http://nulody.design",
  "main": "dist/tokens.js",
  "repository": "https://github.com/nulogy/design-system.git",
  "author": "Nulogy <info@nulogy.com> (https://github.com/nulogy)",
  "license": "MIT",
  "private": false,
  "scripts": {
    "test": "jest",
    "build": "babel src --out-dir dist --ignore *.test*",
    "watch": "yarn build --watch"
  },
  "devDependencies": {
    "babel-cli": "^6.26.0",
    "babel-preset-env": "^1.7.0"
  },
  "dependencies": {
    "babel-jest": "22.4.4",
    "jest": "22.4.4"
  },
  "jest": {
    "testEnvironment": "node"
  },
  "babel": {
    "presets": [
      [
        "env",
        {
          "targets": {
            "node": "current"
          }
        }
      ]
    ]
  }
}
