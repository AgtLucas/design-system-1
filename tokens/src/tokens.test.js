import tokens from './tokens';

describe('tokens', () => {
  it('is valid js', () => {
    expect(tokens).toMatchSnapshot();
  });
});