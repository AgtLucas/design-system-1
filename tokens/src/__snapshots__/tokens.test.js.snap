// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tokens matches snapshot 1`] = `
Object {
  "borderRadius": "3px",
  "breakpoints": Array [
    640,
    1024,
  ],
  "colour": Object {
    "base": "#0E77D2",
    "black": "#03101A",
    "blue": Object {
      "200": "#F0F3F5",
      "300": "#DDE8ED",
      "400": "#C2E0F0",
      "500": "#63B5E8",
      "600": "#0E77D2",
      "700": "#054CA3",
      "800": "#1B2B4D",
      "base": "#0E77D2",
    },
    "error": "#D13D3D",
    "green": Object {
      "200": "#EBF5F3",
      "300": "#C1E8E0",
      "400": "#8DD6C8",
      "500": "#10B297",
      "600": "#008763",
      "700": "#02613C",
      "800": "#053A1F",
      "base": "#008763",
    },
    "neutral": Object {
      "100": "#FFFFFF",
      "200": "#F7F7F7",
      "300": "#EBEBEB",
      "400": "#C7CED4",
      "500": "#98A9B8",
      "600": "#607180",
      "700": "#475866",
      "800": "#203140",
      "900": "#03101A",
    },
    "red": Object {
      "200": "#F7EEED",
      "300": "#F2CECB",
      "400": "#F2B2AE",
      "500": "#DB807D",
      "600": "#D13D3D",
      "700": "#9E131A",
      "800": "#78060F",
      "base": "#D13D3D",
    },
    "success": "#008763",
    "warning": "#FDCF00",
    "white": "#FFFFFF",
    "yellow": Object {
      "200": "#F5F3F0",
      "300": "#F2EADA",
      "400": "#FAE5AF",
      "500": "#FCDA7B",
      "600": "#FDCF00",
      "700": "#CEA10C",
      "800": "#B3751E",
      "base": "#FDCF00",
    },
  },
  "font": Object {
    "family": Object {
      "mono": "'IBM Plex Mono', monospace",
      "regular": "'IBM Plex Sans', sans",
    },
    "lineHeight": Object {
      "regular": 1.5,
      "smaller": 1,
    },
    "size": Object {
      "large": 20,
      "larger": 24,
      "largest": 28,
      "medium": 16,
      "pageTitle": 28,
      "sectionTitle": 24,
      "small": 14,
      "smaller": 12,
      "subsectionTitle": 20,
    },
    "weight": Object {
      "bold": 700,
      "medium": 500,
      "normal": 400,
    },
  },
  "mediaQueries": Object {
    "max": Object {
      "large": "@media screen and (max-width: 1024px)",
      "medium": "@media screen and (max-width: 640px)",
    },
    "min": Object {
      "large": "@media screen and (min-width: 1025px)",
      "medium": "@media screen and (min-width: 641px)",
    },
  },
  "name": "Nulogy Design System",
  "radius": Object {
    "large": 6,
    "medium": 4,
    "small": 2,
  },
  "space": Object {
    "half": 4,
    "x1": 8,
    "x2": 16,
    "x3": 24,
    "x4": 32,
    "x6": 48,
    "x8": 64,
  },
}
`;
